### PR TITLE
Add integration test harness for plugins

### PR DESCRIPTION
**Pull request by rmuir**
Similar to es-core integration tests, in `verify` we unzip ES, install the plugin, run any rest tests it might have against the external cluster, then shut down.

Even if a plugin doesn't have any rest tests, at least we install and start ES with it...

### CODE REVIEW

**Comment by s1monw** [comment]
LGTM - did you just sneak in our new build system... good old ANT

File: dev-tools/src/main/resources/ant/integration-tests.xml [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+              value="-Des.node.name=smoke_tester -Des.cluster.name=prepare_release
+                     -Des.discovery.zen.ping.multicast.enabled=false -Des.script.inline=on
+                     -Des.script.indexed=on -Des.path.repo=${integ.repo.home}"/>
+
+    <!-- execute -->
+    <echo>Starting up external cluster...</echo>
+    <run-script dir="${integ.home}" script="bin/elasticsearch" args="${integ.args}" spawn="true"/>
+
+    <waitfor maxwait="3" maxwaitunit="minute" checkevery="500">
+      <http url="http://127.0.0.1:9200"/>
```
Comment (imotov) [inline dev-tools/src/main/resources/ant/integration-tests.xml]: Would it be possible to make this configurable or at least move it to some other (non-default) port?

Comment (rmuir) [inline dev-tools/src/main/resources/ant/integration-tests.xml]: Yes but its not related to this change.

Comment (rmuir) [inline dev-tools/src/main/resources/ant/integration-tests.xml]: https://github.com/elastic/elasticsearch/issues/12063

**Pull request merged**